### PR TITLE
#77 - Added a shader/mat that pushes water surface to top of geometry

### PR DESCRIPTION
**Pull request by huwb**
The changes to main.unity should be reverted before merging this, but creating this now for discussion.

### CODE REVIEW

**Comment by moosichu** [comment]
Will check this out this weekend, looks interesting!

**Comment by huwb** [comment]
Closing - this PR is now pushed separately in 29f404992177cbd9226bf21622b77d9965badb44 .

@GeorgeDziov , thanks for the comment.

It took me a very long time to get around to it but i finally made changes that fix this issue.

Firstly its worth checking my latest post on #77 for the updates to how this kind of shape input works now.

Now I've made it so the displacement from a shape input can be reported through to the ocean system, which means the ocean tile bounding boxes will be enlarged to capture the displacement and the tiles should not get culled (which is what you see in the video above).

Long story short, if you open boat.unity on master then you'll see the Register Anim Waves Input script on the wave particle has the Max Displacement Vertical set to 5.5, which is the amplitude of the wave particle. This means the ocean tile bounds should always encompass the wave particle and should fix the issue.
